### ISSUE TITLE
Using IDC in other datasets

### ISSUE THREAD

**Issue by YichenShen0103** [XLearning-SCU/2024-NeurIPS-IDC#4]
您好，非常感谢您对论文实验代码的开源，这是一项非常有启发性的工作，我正在进行一项类似的研究工作并希望在另外的数据集上测试IDC的性能来作为我们的参照，我注意到您使用了与训练的TCL模型，但是您在仓库中引用的TCL模型参数（包括若干层的尺寸等）是不同于原版的TCL的，这使得我们无法通过训练原版TCL得到可以被读取的参数，如果方便的话，可以分享您的预训练代码吗？

Thank you very much for sharing the code for your paper. It is a truly inspiring work. I am currently conducting a similar line of research and would like to evaluate the performance of IDC on another dataset as a reference.
I noticed that your implementation uses a pre-trained TCL model; however, the TCL model architecture referenced in your repository (including the dimensions of several layers) appears to differ from the original TCL design. As a result, we are unable to train the original TCL model to produce state dictionary compatible with your implementation.
If possible, could you kindly share the code you used to pre-train the TCL model?

**Comment by Tristan-Liu-13** [comment]
IDC核心的就是sample_selection.py和fine-tune.py，对于预训练的网络结构应该没有特殊要求。可以用原版TCL的去训练，然后将IDC代码的网络对应部分改成他的。
然后我是直接使用的TCL作者提供的checkpoint，没有去训练TCL。当时加载checkpoint是应该是改了两层的（好像就是我下面图里的去掉了nn.BatchNorm1d(512)），因为原版TCL在github上的是多卡训ImageNet的代码，其他数据集都是单卡训练的，有点不一样。

<img width="582" height="717" alt="Image" src="https://github.com/user-attachments/assets/96b7f0c1-50a7-4ccf-98b3-b1573aef29b6" />

<img width="1117" height="769" alt="Image" src="https://github.com/user-attachments/assets/4739c3c9-e945-4ad4-a85e-fb5dc97074f8" />
